pyproject.toml

```diff
@@ -172,3 +174,78 @@ pybamm = [
 
 [tool.setuptools.packages.find]
 include = ["pybamm", "pybamm.*"]
+# TODO: remove once https://github.com/pybamm-team/PyBaMM/issues/3480 is resolved
+exclude = ["pybind11*"]
+
+[tool.ruff]
+extend-include = ["*.ipynb"]
+extend-exclude = ["__init__.py"]
+
+[tool.ruff.lint]
+extend-select = [
+  # "B",           # flake8-bugbear
+  # "I",           # isort
+  # "ARG",         # flake8-unused-arguments
+  # "C4",          # flake8-comprehensions
+  # "ICN",         # flake8-import-conventions
+  # "ISC",         # flake8-implicit-str-concat
+  # "PGH",         # pygrep-hooks
+  # "PIE",         # flake8-pie
+  # "PL",          # pylint
+  # "PT",          # flake8-pytest-style
+  # "PTH",         # flake8-use-pathlib
+  # "RET",         # flake8-return
+  "RUF",         # Ruff-specific
+  # "SIM",         # flake8-simplify
+  # "T20",         # flake8-print
+  # "UP",          # pyupgrade
+  "YTT",         # flake8-2020
+]
+ignore = [
+  "E741",        # Ambiguous variable name
+  "RUF012",      # Mutable class attributes should be annotated with `typing.ClassVar`
+  "SIM108",      # Use ternary operator
+  "ARG001",      # Unused function argument:
+  "ARG002",      # Unused method arguments
+  "PLR2004",     # Magic value used in comparison
+  "PLR0915",     # Too many statements
+  "PLR0913",     # Too many arguments
+  "PLR0912",     # Too many branches
+  "RET504",      # Unnecessary assignment
+  "RET505",      # Unnecessary `else`
+  "RET506",      # Unnecessary `elif`
+  "B018",        # Found useless expression
+  "RUF002",      # Docstring contains ambiguous
+]
+
+[tool.ruff.lint.per-file-ignores]
+"tests/*" = ["T20"]
+"docs/*" = ["T20"]
+"examples/*" = ["T20"]
+"**.ipynb" = ["E402", "E703"]
+
+# NOTE: currently used only for notebook tests with the nbmake plugin.
+[tool.pytest]
+# Use pytest-xdist to run tests in parallel by default, exit with
+# error if not installed
+required_plugins = [
+  "pytest-xdist",
+]
+addopts = [
+  "-nauto",
+  "-v",
+]
+testpaths = [
+  "docs/source/examples/",
+]
+console_output_style = "progress"
+
```

Comment:
I think the `norecursedirs` configuration value should do it: https://docs.pytest.org/en/7.1.x/reference/reference.html#confval-norecursedirs

Reply:
This surprisingly did not work (it should have according to the docs). Ignoring pybind11 in `addopts` works for me locally.

Reply:
I think it failed again on the latest commit with `addopts`, up to you if you want to try another option or wait for #3480 to be completed – I am almost done!

Reply:
Turns out, this wasn't a `pytest`+`pybind11` issue. I was specifying `pytest`'s configuration under `[tool.pytest]`, but it should have been `[tool.pytest.ini_options]`.